test/json/component_status_list.json

```diff
@@ -0,0 +1,52 @@
+{
+  "kind": "ComponentStatusList",
+  "apiVersion": "v1",
+  "metadata": {
+    "selfLink": "/api/v1/componentstatuses"
+  },
+  "items": [
+    {
+      "metadata": {
+        "name": "controller-manager",
+        "selfLink": "/api/v1/namespaces/componentstatuses/controller-manager",
+        "creationTimestamp": null
+      },
+      "conditions": [
+        {
+          "type": "Healthy",
+          "status": "Unknown",
+          "error": "Get http://127.0.0.1:10252/healthz: dial tcp 127.0.0.1:10252: connection refused"
+        }
+      ]
+    },
+    {
+      "metadata": {
+        "name": "scheduler",
+        "selfLink": "/api/v1/namespaces/componentstatuses/scheduler",
+        "creationTimestamp": null
+      },
+      "conditions": [
+        {
+          "type": "Healthy",
+          "status": "Unknown",
+          "error": "Get http://127.0.0.1:10251/healthz: dial tcp 127.0.0.1:10251: connection refused"
+        }
+      ]
+    },
+    {
+      "metadata": {
+        "name": "etcd-0",
+        "selfLink": "/api/v1/namespaces/componentstatuses/etcd-0",
```

Comment:
> b. /api/v1/namespaces/{namespace}/componentstatuses
> this doesn't make sense, it should return everything for every namespace? can you also open a question on k8s side if this is unclear?

This has already been shipped in v1 so they won't remove it (and anyway it's already out there in existing deployments). Kubeclient only "blindly" exposes entry points for entities and what they return, that said @abonas do you see any reason to block this PR?